browser/lib/util/webstorage.js

```diff
@@ -9,25 +10,26 @@ var WebStorage = (function() {
 	 * somewhat roundabout way. (If unsupported or no window object,
 	 * will throw on accessing a property of undefined) */
 	try {
-		window.sessionStorage.setItem(test, test);
-		window.sessionStorage.removeItem(test);
+		globalObject.sessionStorage.setItem(test, test);
+        globalObject.sessionStorage.removeItem(test);
 		sessionSupported = true;
 	} catch(e) {
 		sessionSupported = false;
 	}
 
 	try {
-		window.localStorage.setItem(test, test);
-		window.localStorage.removeItem(test);
+        globalObject.localStorage.setItem(test, test);
+        globalObject.localStorage.removeItem(test);
 		localSupported = true;
 	} catch(e) {
 		localSupported = false;
 	}
 
-	function WebStorage() {}
+
+    function WebStorage() {}
 
 	function storageInterface(session) {
-		return session ? window.sessionStorage : window.localStorage;
+		return session ? globalObject.sessionStorage : globalObject.localStorage;
```

Comment:
When I wrote the patch around WebStorage, I wasn't sure if we needed it for anything.  So it made more sense to make the webStorage check the global object, this also allows this code to work inside browser webworkers as a webworker also has no `window` variable; but does have a `global` object with webstorage attached; so I was attempting to not only fix it so if NS needed it, but make Ably also be a bit more WebWorker friendly.  :-D

1. Yes, once something is created in NativeScript it exists forever or until GC'd (just like Node). 
2. Yes, NS has support for both xhr and WebSocket.  Persisting the info might be useful in the cases where WebSocket is always failing, but doesn't ably by default try WS then XHR; so in most cases I wouldn't think this info would not need to be persisted between app runs.